### DISCUSSION TITLE
How to change spelling?

### DISCUSSION THREAD

**Discussion by dudicoco** [catppuccin/nvim#671] [Q&A]
Currently misspelled words are underlined in the same color as the text, I would like them to be underlined in red, how can that be customized?

thanks

**Comment by mrtnvgr** [comment]
```lua
require("catppuccin").setup {
    custom_highlights = function(colors)
        local highlights = {}

        local spell_options = { style = {"underline"}, fg = colors.red }
        local spell_groups = {"SpellBad", "SpellCap", "SpellLocal", "SpellRare"}
        for _, v in ipairs(spell_groups) do
            highlights[v] = spell_options
        end

        return highlights
    end
}
```

```help
SpellBad	Word that is not recognized by the spellchecker. |spell|
		Combined with the highlighting used otherwise.
							*hl-SpellCap*
SpellCap	Word that should start with a capital. |spell|
		Combined with the highlighting used otherwise.
							*hl-SpellLocal*
SpellLocal	Word that is recognized by the spellchecker as one that is
		used in another region. |spell|
		Combined with the highlighting used otherwise.
							*hl-SpellRare*
SpellRare	Word that is recognized by the spellchecker as one that is
		hardly ever used. |spell|
		Combined with the highlighting used otherwise.
```

**Reply by dudicoco** [reply]
thanks @mrtnvgr!

I would like the underline to be colored rather than the whole word, i've tried using `guisp` instead of `fg` but got an error that it's unrecognized by catppuccin.
Is there a way to achieve this?

**Reply by mrtnvgr** [reply]
>  Is there a way to achieve this?

You should use `sp` instead of `guisp`, because `nvim_set_hl` function treats highlight group options differently:

![](https://i.stack.imgur.com/sbFOV.png)

Remember that `gui*` options work only in GUI-based environments (e.g. `neovide`) 🙂

**Reply by dudicoco** [reply]
I see, so is there a way to color the underline in the terminal?

**Reply by mrtnvgr** [reply]
> I see, so is there a way to color the underline in the terminal?

I couldn't find any, which terminal do you use?

**Reply by dudicoco** [reply]
kitty

**Reply by mrtnvgr** [reply]
Did you try `sp`? ~maybe it will work suddenly, who knows~ If you really need this, I can highly recommend you neovide. For seemless experience, you can enable "swallowing" for neovide in your window/tiling manager settings, alias `nvim` to `neovide` and set a gui font for neovim :)

**Reply by dudicoco** [reply]
Thanks for the suggestion but I have no interest in moving to neovide :)
